### PR TITLE
BL-2477: Add inline, pop-up ckeditor for bold, underline, italic

### PR DESCRIPTION
**Pull request by phillip-hopper**
Only 5 files were modified, `bloomBootstrap.js`, `bloomEditing.js`, `editMode.css`, `editMode.less` and `ProjectContext.cs`. The other 76 files that were added are part of ckeditor.

Only the files necessary to run ckeditor in Firefox have been included. This configuration does not work in Chrome.

### CODE REVIEW

File: src/BloomExe/web/EnhancedImageServer.cs [hunk 1]
```diff
@@ -568,6 +569,12 @@ private bool ProcessCssFile(IRequestInfo info, string localPath)
 			// if still not found, and localPath is an actual file path, use it
 			if (string.IsNullOrEmpty(path) && File.Exists(localPath)) path = localPath;
 
+			// check if this is a path in BloomBrowserUI
+			if (string.IsNullOrEmpty(path))
+			{
+				path = FileLocator.GetFileDistributedWithApplication("BloomBrowserUI", localPath);				
+			}
+
```
Comment (JohnThomson) [inline src/BloomExe/web/EnhancedImageServer.cs]: Shouldn't we just configure the _fileLocator used above to know about this possible location?

**Comment by JohnThomson** [comment]
I'm rather dismayed that we need to add such a mass of files to get one little control, but I suppose JohnH hopes to use this for more stuff eventually. Is there (or could we create?) a minified version of the ckeditor stuff to both speed things up and maybe reduce our installer size a bit?

**Comment by JohnThomson** [comment]
I'm also a bit dismayed by the size and complexity of editor_gecko.less. I think there should, at least, be a comment at the head of the file summarizing what it is used for and how (and anything you can say about why) we are modifying the standard ckeditor appearance. Also, since I think we only use the ckeditor control on divs, I'm surprised that we have to modify it for so many other kinds of thing.

**Comment by JohnThomson** [comment]
The name editor_gecko.less doesn't help me very much. I would guess such a file had to do with tricks we needed to make editing in general work in gecko, whereas I gather the real point is to tweak the appearance of the ckeditor control. Is there gecko-specific stuff in it? Would ckeditor_tweaks.less express the purpose better?

**Comment by JohnThomson** [comment]
The issue calls for this new control to appear only when a range is selected. If there is something to make this so (or to make it appear at all) I haven't found it. I'm also a bit surprised that there isn't any need for this to interact with our existing code for handling ctrl-B etc. Probably not a problem, but I thought I'd ask...

**Comment by phillip-hopper** [comment]
I already removed most of the files.  66 of the 77 remaining files are localizations.

**Comment by phillip-hopper** [comment]
I haven't found anything that only shows the toolbar if a range is selected.

**Comment by phillip-hopper** [comment]
Ctrl+B, I and U still work just as they are supposed to.

**Comment by phillip-hopper** [comment]
I restored the factory `editor_gecko.css`, and instead of adding another less file, I was able to achieve the same result by adding a few lines to `editMode.less`.

**Comment by JohnThomson** [comment]
Do we need 66 localizations of this editor? Bloom itself doesn't have that many (yet). Admittedly they are fairly small...

**Comment by JohnThomson** [comment]
Regarding editor_gecko.less: have you committed this yet? The overall comment still says that it is the one added file that is not part of ckeditor, and I don't see any changes at all to ediitMode.less.

**Comment by JohnThomson** [comment]
I'm concerned that you could not achieve "show only when a range is selected." My impression is that @hatton thinks this is fairly important, to avoid as far as possible complicating the UI except when someone really might be trying to change formatting. I'm not sure exactly how much effort it is worth...maybe we can talk about it if you're able to make standup.

**Comment by phillip-hopper** [comment]
Pull request updated now.

**Comment by hatton** [comment]
Showing only when text is selected is mandatory.

**Comment by phillip-hopper** [comment]
It now shows only when selected.

**Comment by JohnThomson** [comment]
I think the only remaining quibble is that the commit description is no longer accurate about editor_gecko.css. This is now just one of the unmodified ckeditor files, right?

**Comment by phillip-hopper** [comment]
Commit description updated.

**Pull request merged**